### PR TITLE
Don't always assume plugins should be activated

### PR DESCRIPTION
**Pull request by jefferai**
When running test controllers from e.g. Terraform tests we don't have the external plugins built in, so don't always include them. Additionally, add the loopback plugin to the test controller.

### CODE REVIEW

File: internal/servers/controller/testing.go [hunk 1]
```diff
@@ -414,6 +416,17 @@ func NewTestController(t *testing.T, opts *TestControllerOpts) *TestController {
 		t.Fatal(err)
 	}
 
+	tc.b.DevLoopbackHostPluginId = DefaultTestPluginId
+	plg := plugin.NewWrappingPluginClient(plugin.NewLoopbackPlugin())
+	plugOpts := []hostplugin.Option{
+		hostplugin.WithDescription("Provides an initial loopback host plugin in Boundary"),
+		hostplugin.WithPublicId(tc.b.DevLoopbackHostPluginId),
+	}
+	if _, err = tc.b.RegisterHostPlugin(ctx, "loopback", plg, plugOpts...); err != nil {
```
Comment (talanknight) [inline internal/servers/controller/testing.go:425]: Please remove the part of this file which adds the loopback plugin already (see the section close to `CreateDevDatabase` in TestControllerConfig)

File: internal/servers/controller/controller.go [hunk 2]
```diff
@@ -119,32 +123,38 @@ func New(ctx context.Context, conf *Config) (*Controller, error) {
...
-		external_host_plugins.WithLogger(hclog.NewNullLogger()))
-	if err != nil {
-		return nil, fmt.Errorf("error creating aws host plugin")
-	}
-	conf.ShutdownFuncs = append(conf.ShutdownFuncs, awsCleanup)
-	if _, err := conf.RegisterHostPlugin(ctx, "aws", awsSvcClient, hostplugin.WithDescription("Built-in AWS host plugin")); err != nil {
-		return nil, fmt.Errorf("error registering aws host plugin: %w", err)
+	for _, enabledPlugin := range c.enabledPlugins {
+		switch enabledPlugin {
+		case base.EnabledPluginHostAzure:
```
Comment (talanknight) [inline internal/servers/controller/controller.go:128]: These 2 case statements seem very similar.  Consider containing the plugin information (like the name and... is that it?) with the EnabledPluginHost type so there is a centralized place for that info and this code can avoid being duplicated.

**Review by talanknight — CHANGES_REQUESTED** [review]

File: internal/servers/controller/testing.go (hunk 1 above)
Comment (jefferai) [reply internal/servers/controller/testing.go:425]: TIL we already have loopback in test controllers!

File: internal/servers/controller/controller.go (hunk 2 above)
Comment (jefferai) [reply internal/servers/controller/controller.go]: Good idea, fixed!

**Review by talanknight — APPROVED** [review]

**Pull request merged**